third-party/folly/folly/synchronization/Baton.h

```diff
@@ -256,7 +250,9 @@ class Baton {
       const std::chrono::time_point<Clock, Duration>& deadline,
       const WaitOptions& opt) noexcept {
 
-    switch (detail::spin_pause_until(deadline, opt, [ROCKSDB_THIS_LAMBDA_CAPTURE] { return ready(); })) {
+    // Avoid implicit/explicit capture 'this' because C++ standards change
+    const auto thiz = this;
+    switch (detail::spin_pause_until(deadline, opt, [thiz] { return thiz->ready(); })) {
```

Comment:
`[this] { return ready();` should be enough, no ?

Reply:
You're right. Unlike [=,this], [this] should standard (C++11 and on). Got my wires crossed

Reply:
@pdillinger Me too... maybe removing all implicit captures can just fix the issue without ugly preprocessor define. I wanted to give a try before tomorrow but I saw your PR :)